topics/basics/testing_plugins/integration_tests/integration_tests_intro.md

```diff
@@ -0,0 +1,298 @@
+  }
+
+sourceSets {
+  create("intTest") {
+    compileClasspath += sourceSets.main.get().output
+    runtimeClasspath += sourceSets.main.get().output
+  }
+}
+
+val intTestImplementation by configurations.getting {
+  extendsFrom(configurations.testImplementation.get())
+}
+
+dependencies {
+  intTestImplementation("org.junit.jupiter:junit-jupiter:5.7.1")
+  intTestImplementation("org.kodein.di:kodein-di-jvm:7.20.2")
+  intTestImplementation("org.jetbrains.kotlinx:kotlinx-coroutines-core-jvm:1.10.1")
+}
+
+val integrationTest = task<Test>("integrationTest") {
+  description = "Runs integration tests."
+  group = "verification"
+  testClassesDirs = sourceSets["intTest"].output.classesDirs
+  classpath = sourceSets["intTest"].runtimeClasspath
+  dependsOn("buildPlugin")
+  systemProperty("path.to.build.plugin", tasks.buildPlugin.get().archiveFile.get().asFile.absolutePath)
+  useJUnitPlatform()
+}
+```
+
+The following dependencies are required:
+
+* `testFramework(TestFrameworkType.Starter)` will add all required dependencies for writing integration tests - Starter and Driver frameworks.
+* `org.kodein.di:kodein-di-jvm` is a dependency injection framework used by Starter for configuration.
+* `org.jetbrains.kotlinx:kotlinx-coroutines-core-jvm` is required for Starter framework which is implemented using Kotlin coroutines.
+
+This configuration does the following:
+
+* Imports test implementation dependencies to intTest implementation dependencies
+* Defines new test source roots in `src/intTest`
+* Creates a new task `integrationTest` in `verification` group
+* Makes the test task depend on `buildPlugin`, ensuring plugin is built before tests run.
+* To test a plugin, the Starter framework needs to know where to find the plugin distribution for installation in the IDE.
+`path.to.build.plugin` system property points to the plugin distribution file.
+* Enables JUnit Platform for test execution.
+
+For more details about configuring integration tests, please refer to [Gradle docs](https://docs.gradle.org/current/userguide/java_testing.html#sec:configuring_java_integration_tests).
+
+> Note: Driver and UI components are in the experimental state and subject to API changes.
+>
+{style="note"}
+
+## Creating the First Integration Test
+
+Now that the configuration is complete, it's time to write the first integration test, which will:
+
+* Start the IDE with the plugin installed.
+* Wait for all background processes to complete.
+* Perform a shutdown.
+
+Create a new Kotlin file in `src/intTest/kotlin` with the following code:
+
+```kotlin
+class PluginTest {
+  @Test
+  fun simpleTestWithoutProject() {
+    Starter.newContext(testName = "testExample", TestCase(IdeProductProvider.IC, projectInfo = NoProject).withVersion("2024.3")).apply {
+      val pathToPlugin = System.getProperty("path.to.build.plugin")
+      PluginConfigurator(this).installPluginFromPath(Path(pathToPlugin))
+    }.runIdeWithDriver().useDriverAndCloseIde {
+    }
+  }
+}
+```
```

Comment:
```suggestion
```kotlin
class PluginTest {
  @Test
  fun simpleTestWithoutProject() {
    Starter.newContext(
      testName = "testExample",
      TestCase(IdeProductProvider.IC, projectInfo = NoProject)
        .withVersion("2024.3")
    ).apply {
      val pathToPlugin = System.getProperty("path.to.build.plugin")
      PluginConfigurator(this).installPluginFromPath(Path(pathToPlugin))
    }.runIdeWithDriver().useDriverAndCloseIde {
    }
  }
}
```
```

We have 70 characters of width, otherwise readers have to scroll. Let's minimize it.